packages/splits-sdk-react/src/hooks/warehouse.ts

```diff
@@ -0,0 +1,107 @@
+import { Log } from 'viem'
+import { useCallback, useContext, useState } from 'react'
+import {
+  WarehouseWithdrawConfig,
+  WarehouseBatchWithdrawConfig,
+} from '@0xsplits/splits-sdk'
+
+import { SplitsContext } from '../context'
+import { ContractExecutionStatus, RequestError } from '../types'
+import { getSplitsClient } from '../utils'
+
+export const useWithdrawWarehouse = (): {
+  withdrawWarehouse: (
+    arg0: WarehouseWithdrawConfig,
+  ) => Promise<Log[] | undefined>
+  status?: ContractExecutionStatus
+  txHash?: string
+  error?: RequestError
+} => {
+  const context = useContext(SplitsContext)
+  const splitsClient = getSplitsClient(context).warehouse
+
+  if (!splitsClient) throw new Error('Invalid chain id for splits warehouse')
+
+  const [status, setStatus] = useState<ContractExecutionStatus>()
+  const [txHash, setTxHash] = useState<string>()
+  const [error, setError] = useState<RequestError>()
+
+  const withdrawWarehouse = useCallback(
+    async (argsDict: WarehouseWithdrawConfig) => {
+      try {
+        setStatus('pendingApproval')
+        setError(undefined)
+        setTxHash(undefined)
+
+        const { txHash: hash } =
+          await splitsClient.submitWithdrawTransaction(argsDict)
+
+        setStatus('txInProgress')
+        setTxHash(hash)
+
+        const events = await splitsClient.getTransactionEvents({
+          txHash: hash,
+          eventTopics: splitsClient.eventTopics.withdraw,
+        })
+
+        setStatus('complete')
+
+        return events
+      } catch (e) {
+        setStatus('error')
+        setError(e)
+      }
+    },
+    [splitsClient],
+  )
+
+  return { withdrawWarehouse, status, txHash, error }
+}
+export const useBatchWithdrawWarehouse = (): {
+  batchWithdrawWarehouse: (
+    arg0: WarehouseBatchWithdrawConfig,
+  ) => Promise<Log[] | undefined>
+  status?: ContractExecutionStatus
+  txHash?: string
+  error?: RequestError
+} => {
+  const context = useContext(SplitsContext)
+  const splitsClient = getSplitsClient(context).warehouse
+
+  if (!splitsClient) throw new Error('Invalid chain id for splits warehouse')
+
+  const [status, setStatus] = useState<ContractExecutionStatus>()
+  const [txHash, setTxHash] = useState<string>()
+  const [error, setError] = useState<RequestError>()
+
+  const batchWithdrawWarehouse = useCallback(
+    async (argsDict: WarehouseBatchWithdrawConfig) => {
+      try {
+        setStatus('pendingApproval')
+        setError(undefined)
+        setTxHash(undefined)
+
+        const { txHash: hash } =
+          await splitsClient.submitBatchWithdrawTransaction(argsDict)
+
+        setStatus('txInProgress')
+        setTxHash(hash)
+
+        const events = await splitsClient.getTransactionEvents({
+          txHash: hash,
+          eventTopics: splitsClient.eventTopics.withdraw,
+        })
+
+        setStatus('complete')
+
+        return events
+      } catch (e) {
+        setStatus('error')
+        setError(e)
+      }
```

Comment:
nit: wonder if we can pull this pattern out somehow. getting reused so much.

anyway, sorry for all these comments that aren't specific to this pr. first time i'm taking a good look at this react sdk code in a while.